autopush/main_argparse.py

```diff
@@ -0,0 +1,258 @@
+    parser.add_argument('--s3_bucket', help='OBSOLETE')
+    parser.add_argument('--senderid_expry', help='OBSOLETE')
+    parser.add_argument('--proxy_protocol', help="OBSOLETE")
+    # old APNs args
+    parser.add_argument('--apns_enabled', help="OBSOLETE")
+    parser.add_argument('--apns_sandbox', help="OBSOLETE")
+    parser.add_argument('--apns_cert_file', help="OBSOLETE")
+    parser.add_argument('--apns_key_file', help="OBSOLETE")
+
+
+def _add_external_router_args(parser):
+    """Parses out external router arguments"""
+    # GCM
+    parser.add_argument('--gcm_enabled', help="Enable GCM Bridge",
+                        action="store_true", default=False,
+                        env_var="GCM_ENABLED")
+    label = "GCM Router:"
+    parser.add_argument('--gcm_ttl', help="%s Time to Live" % label,
+                        type=int, default=60, env_var="GCM_TTL")
+    parser.add_argument('--gcm_dryrun',
+                        help="%s Dry run (no message sent)" % label,
+                        action="store_true", default=False,
+                        env_var="GCM_DRYRUN")
+    parser.add_argument('--gcm_collapsekey',
+                        help="%s string to collapse messages" % label,
+                        type=str, default="simplepush",
+                        env_var="GCM_COLLAPSEKEY")
+    parser.add_argument('--senderid_list', help='GCM SenderIDs/auth keys',
+                        type=str, default="{}")
+    # FCM
+    parser.add_argument('--fcm_enabled', help="Enable FCM Bridge",
+                        action="store_true", default=False,
+                        env_var="FCM_ENABLED")
+    label = "FCM Router:"
+    parser.add_argument('--fcm_ttl', help="%s Time to Live" % label,
+                        type=int, default=60, env_var="FCM_TTL")
+    parser.add_argument('--fcm_dryrun',
+                        help="%s Dry run (no message sent)" % label,
+                        action="store_true", default=False,
+                        env_var="FCM_DRYRUN")
+    parser.add_argument('--fcm_collapsekey',
+                        help="%s string to collapse messages" % label,
+                        type=str, default="simplepush",
+                        env_var="FCM_COLLAPSEKEY")
+    parser.add_argument('--fcm_auth', help='Auth Key for FCM',
+                        type=str, default="")
+    parser.add_argument('--fcm_senderid', help='SenderID for FCM',
+                        type=str, default="")
+    # Apple Push Notification system (APNs) for iOS
+    # credentials consist of JSON struct containing a channel type
+    # followed by the settings,
+    # e.g. {'firefox':{'cert': 'path.cert', 'key': 'path.key',
+    #                  'sandbox': false}, ... }
+    parser.add_argument('--apns_creds', help="JSON dictionary of "
+                                             "APNS settings",
+                        type=str, default="",
+                        env_var="APNS_CREDS")
+    # UDP
```

Comment:
I could nuke it in this PR but I'd prefer logging a separate issue for this. This change is so large that I'd rather play it safe to keep the main -> main_args delta as straightforward as possible